### PR TITLE
feat: add comment to each request

### PR DESCRIPTION
**Pull request by LeoMavri**
I'll handle tests in the morning.

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -48,10 +52,26 @@ class TornAPI {
...
+        if (this.comment !== comment) {
+            this.comment = comment;
+
+            // when the comment changes, reset all the internals
+            this._torn = null;
+            this._user = null;
+            this._faction = null;
+            this._property = null;
+            this._itemmarket = null;
+            this._company = null;
```
Comment (jgolla) [inline lib/index.ts:67]: Can we extract this from here and `setKey` into its own routine?

Comment (jgolla) [reply lib/index.ts:67]: Just for more info, `setKey` and `setComment` functions are almost identical. I would like to remove the following from each and put it into its own routine that is called from both the setters.

 ```
            // when the comment changes, reset all the internals
            this._torn = null;
            this._user = null;
            this._faction = null;
            this._property = null;
            this._itemmarket = null;
            this._company = null;
```

So maybe ...

```
private resetInternals(): void {
  this._torn = null;
  this._user = null;
  this._faction = null;
  this._property = null;
  this._itemmarket = null;
  this._company = null;
}
```

**Comment by jgolla** [comment]
Looking this over, I am fine with not doing any tests for this at the moment. There are no existing test for index.ts anyway, but maybe need to add them later for both the key and the comment.

**Comment by LeoMavri** [comment]
This has been handled here [eba36a9](https://github.com/jgolla/torn-api/pull/397/commits/eba36a96fef68fb0819ea9a962d926f1e7d3b77d)

**Pull request merged**